src/legacy/core_plugins/kibana/public/discover/helpers/get_index_pattern_id.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IndexPattern } from '../../../../data/public/index_patterns';
```

Comment:
These functions are pretty generic.
Can you please move them to `src/plugins/data/common/es_query/utils`
It will immediately be available to you on:
```
import { utils } from '../../../../plugins/data/public`;
utils.findIndexPatternById()
```

I would also recomment using `IIndexPattern` for the interface

Reply:
Mind if I do this in a separate PR, I'd optimize for general usage